### PR TITLE
chore: Silence pytest DeprecationWarnings

### LINKED ISSUE #322: Fix pytest warnings
### Feature request

@0dm to paste

### Motivation

_No response_

### PR DESCRIPTION
**Pull request by 0dm**
closes #322 by ignoring DeprecationWarnings within pyproject.toml, since most of the time we can't directly fix this within OpenAdapt (for most of them?)

new env variable: SILENCE_DEPRECATION (true by default) 

before:
image
after:
image

### CODE REVIEW

File: openadapt/config.py [hunk 1]
```diff
@@ -139,6 +141,14 @@ def getenv_fallback(var_name: str) -> str:
 DB_URL = f"sqlite:///{DB_FPATH}"
 DIRNAME_PERFORMANCE_PLOTS = "performance"
 
+with open(ROOT_DIRPATH / "pyproject.toml", "r") as f:
```
Comment (abrichr) [inline openadapt/config.py:144]: Please add a comment explaining this is a workaround for the lack of support in 3.10 that is in 3.11

File: openadapt/config.py [hunk 2]
```diff
@@ -139,6 +141,14 @@ def getenv_fallback(var_name: str) -> str:
...
+    content = f.read()
+    document = tomlkit.parse(content)
+option = ["ignore::DeprecationWarning"] if SILENCE_DEPRECATION else ""  # type: ignore # noqa
+document["tool"]["pytest"]["ini_options"]["filterwarnings"] = option
```
Comment (abrichr) [inline openadapt/config.py:148]: How about:
```
if SILENCE_DEPRECATION:
    document["tool"]["pytest"]["ini_options"]["filterwarnings"] = ["ignore::DeprecationWarning"]
```

**Comment by ayewo** [comment]
Hey @0dm 👋,

@abrichr asked me to review some of the open PRs so I decided to start with yours then slowly work my way to the others.

I left some review comments.

Nice work 💪!

**Comment by 0dm** [comment]
> Hey @0dm 👋,
> 
> @abrichr asked me to review some of the open PRs so I decided to start with yours then slowly work my way to the others.
> 
> I left some review comments.
> 
> Nice work 💪!

Hi,

I don't see any comments, could you click submit on the review?

**Review by ayewo — COMMENTED** [review]
My bad! I forgot to hit the submit review button 😀.

File: openadapt/config.py [hunk 3]
```diff
@@ -171,6 +173,14 @@ def persist_env(var_name: str, val: str, env_file_path: str = ENV_FILE_PATH) ->
...
+with open(ROOT_DIRPATH / "pyproject.toml", "w") as f:
+    f.write(tomlkit.dumps(document))
+
```
[the remark this answers is not in the document]
Comment (0dm) [reply openadapt/config.py:183]: if we toggle it off in config then it'll write the changes to pyproject

File: openadapt/config.py [hunk 4]
```diff
@@ -85,6 +86,7 @@
         "children",
     ],
     "PLOT_PERFORMANCE": True,
+    "SILENCE_DEPRECATION": True,
```
[the remark this answers is not in the document]
Comment (0dm) [reply openadapt/config.py]: I think this is a more appropriate. Thanks.

[the remark this answers is not in the document]
Comment (0dm) [reply openadapt/config.py]: see [466f8cc](https://github.com/OpenAdaptAI/OpenAdapt/pull/407/commits/466f8cc0250e3d107af4a3e4f7579b1656d34b0e)

**Review by ayewo — APPROVED** [review]
LGTM.

**Pull request merged**